src/gui/PluginBrowser.cpp

```diff
@@ -287,4 +291,30 @@ void PluginDescWidget::mousePressEvent( QMouseEvent * _me )
 }
 
 
+void PluginDescWidget::contextMenuEvent(QContextMenuEvent * e)
+{
+	QMenu contextMenu(this);
+
+	contextMenu.addAction(
+		tr("Send to new instrument track"),
+		[=]{ openInNewInstrumentTrack(m_pluginKey.desc->name); }
+	);
+	
+	contextMenu.exec(e->globalPos());
+}
+
+
+void PluginDescWidget::openInNewInstrumentTrack(QString value)
+{
+	TrackContainer* tc = Engine::getSong();
+	
+	InstrumentTrack * it = dynamic_cast<InstrumentTrack*>(
+			Track::create(Track::InstrumentTrack,
+							tc));
+	InstrumentLoaderThread *ilt = new InstrumentLoaderThread(
+				this, it, value);
```

Comment:
```suggestion
	auto it = dynamic_cast<InstrumentTrack*>(Track::create(Track::InstrumentTrack, tc));
	auto ilt = new InstrumentLoaderThread(this, it, value);
```

Reply:
Also it looks like this `ilt` pointer is hanging. What frees it?

Reply:
InstrumentLoaderThread is a QThread so Qt should free it.  TrackContainerView.cpp does the same thing.

Reply:
@PhysSong @DomClark Can you give some insight on this? Does the loader thread get freed from the heap somewhere?

Reply:
``InstrumentLoaderThread`` is a child of ``this`` (the ``PluginBrowser``), so the thread will be freed when its parent gets deleted.

Reply:
Ah, gotcha! Everything I saw related to QThread wasn't saying anything about it freeing children (or even being allocated on the heap), but if the parent destructs it then cool. 👍